src/library_html5_webgl.js

```diff
@@ -91,6 +91,16 @@ var LibraryHtml5WebGL = {
       renderViaOffscreenBackBuffer: HEAP8[attributes + {{{ C_STRUCTS.EmscriptenWebGLContextAttributes.renderViaOffscreenBackBuffer }}}]
     };
 
+
+#if ASSERTIONS
+    assert(contextAttributes.majorVersion === 1 || contextAttributes.majorVersion === 2, `${contextAttributes.majorVersion} is not a valid value for contextAttributes.majorVersion (should be 1 or 2)`);
+#if MIN_WEBGL_VERSION >= 2
+    assert(contextAttributes.majorVersion === 2, 'Requesting a WebGL context with version 1 is incompatible with linker flag -sMIN_WEBGL_VERSION=2');
+#elif MAX_WEBGL_VERSION == 1
+    assert(contextAttributes.majorVersion === 1, 'Requesting a WebGL context with version 2 requires either -sMIN_WEBGL_VERSION=2 or -sMAX_WEBGL_VERSION=2 linker flag');
```

Comment:
I wonder if we should actually fail in this case?  Even in release mode?

It seems better than just ignoring the invalid values maybe?

Would that break a lot of existing code?   It seems like the asserts are going to break just as many users (just in their debug builds only).

Reply:
If we do stick with asserts maybe we can shorten these errors a little too.

How about: "WebGL2 requested but only WebGL1 is supported (MAX_WEBGL_VERSION is 1)"

Reply:
I thought about the "failing in release mode" as well, but I am just worried that it might break a lot of code that doesn't really care.

For example, ImGui requests a WebGL 2 context (via the `GLFW_CONTEXT_VERSION_MAJOR` API), but does not provide any compiler flag and as result really gets a WebGL 1 (+ extensions because that is the default) and clearly that is enough for ImGui to work. Maybe there is a lot of other code that behave in a similar fashion.

Also note that these changes are **only** in place when calling the `emscripten_webgl_create_context` API. I purposely did not put these checks in `GL.createContext` which gets called by other parts of the code like `Browser.createContext`.

I think I would leave it in debug build only.

Reply:
But the number of folks with broken build are the same either way.  Only breaking their debug builds is still breaking them.

How about we use `warn` here in debug builds, which would mean folks would have time to investivate?  We can then turn the `warn` into an error at some point in the future?

Reply:
sounds like a good idea:

1) how do I implement it?
2) how do I test it?

Thanks

Reply:
```
#if ASSERTIONS
if (...) {
  warm(...);
} 
#endif
```

Testing here is a little harder because getting access to the console messages is tricky... we would have to look at which other tests to this.

Reply:
Ok I will investigate other tests. Thanks for the pointer.

Reply:
I think you might have to do something like in `test/emscripten_console_log_pre.js` in order to detect the warn in code.

Then in the test code would write something like:

```
int main() {
   ...
  EM_ASM(assert(warnOutput == EXPECTED_WARN_OUTPUT));
  ...
}
```

Reply:
I think I'd also be ok with skipping testing in this case, assuming you verify the warnings manually.

Reply:
I am not sure what I am doing wrong, but adding `$warn` to the list of `__deps` does not seem to work:

```
error: undefined symbol: $warn (referenced by emscripten_webgl_do_create_context, referenced by emscripten_webgl_create_context, referenced by root reference (e.g. compiled C/C++ code))
```

and of course not adding it result in:

```
+//report_result?exception:warn is not defined / ReferenceError: warn is not defined
```

any idea?

Reply:
Oh sorry, I thought we had `warn` function but I guess we don't at runtime anyway :)

We have `warnOnce` in the library, but in this case I think you can just do `err(..)` so show the error.  You don't need to depend on `err` since its always available.